federation-integration-testsuite-js/src/matchers/toHaveFetched.ts

```diff
@@ -15,8 +16,8 @@ function prepareHttpOptions(requestUrl: string, requestOpts: RequestInit): Reque
   const headers = new Headers();
   headers.set('Content-Type', 'application/json');
   if (requestOpts.headers) {
-    for (let name in requestOpts.headers) {
-      headers.set(name, requestOpts.headers[name]);
+    for (const [name, value] of new Headers(requestOpts.headers)) {
```

Comment:
Just so I'm understanding, this is necessary — rather [than just `headers.entries()`](https://developer.mozilla.org/en-US/docs/Web/API/Headers/entries) to account for the fact that `headers` might be a `{ [name: string]: string }`?

Reply:
This code hasn't changed in this PR, but I'm wondering why we can't directly initialize headers as `new Headers(requestOpts.headers)`. What is iterating over the passed in headers and adding them one by one adding?

Reply:
Also, the types of `toHaveFetched` are currently broken because the normal `BodyInit` doesn't support arbitrary objects, but here it is always assumed an object is passed in and should be serialized as JSON.